### PR TITLE
docs: Updating the app_cookie_stickiness_policy docs to reflect needing a L…

### LINKED ISSUE #2926: Incorrect documentation for aws_app_cookie_stickiness_policy - elb.id doesn't exist
Example shows

load_balancer = "${aws_elb.lb.id}"

but should be

load_balancer = "${aws_elb.lb.name}"

as id causes a "no such attribute" type error

### PR DESCRIPTION
**Pull request by stack72**
…oadBalancer name NOT Id

Fixes #2926

### CODE REVIEW

**Comment by radeksimko** [comment]
Name and ID are actually [equal](https://github.com/hashicorp/terraform/blob/master/builtin/providers/aws/resource_aws_elb.go#L255), so it does not matter, but it's nice to be explicit in docs. :+1:

File: website/source/docs/providers/aws/r/app_cookie_stickiness_policy.html.markdown [hunk 1]
```diff
@@ -15,20 +15,20 @@ Provides an application cookie stickiness policy, which allows an ELB to wed its
...
+	  lb_protocol = "http"
+    }
 }
 
 resource "aws_app_cookie_stickiness_policy" "foo" {
-	  name = "foo_policy"
-	  load_balancer = "${aws_elb.lb.id}"
-	  lb_port = 80
-	  cookie_name = "MyAppCookie"
+	name = "foo_policy"
```
Comment (radeksimko) [inline website/source/docs/providers/aws/r/app_cookie_stickiness_policy.html.markdown]: This is however wrong, when I tried applying this example, I got an error:

```
1 error(s) occurred:

* aws_app_cookie_stickiness_policy.foo: Error creating AppCookieStickinessPolicy: ValidationError: Policy name cannot contain characters that are not letters, or digits or the dash.
	status code: 400, request id: c96ca9bd-5eed-11e5-b3c8-1b09bd7b9609
```

It would be nice to add validation for `aws_app_cookie_stickiness_policy.name`, but more importantly the example should be fixed - e.g. by replacing underscore with dash.

Comment (stack72) [inline website/source/docs/providers/aws/r/app_cookie_stickiness_policy.html.markdown]: Ah so this is actually an error with app_cookie_stickiness_policy?

Comment (radeksimko) [inline website/source/docs/providers/aws/r/app_cookie_stickiness_policy.html.markdown]: Well, you didn't cause this by your changes :smiley: but the example has invalid name for some reason, so I think we can fix it.

Comment (stack72) [inline website/source/docs/providers/aws/r/app_cookie_stickiness_policy.html.markdown]: Cool - I'll update my PR to include the ValidateFunc and an updated
example. May also check we have acceptance tests that cover it :)

Comment (stack72) [inline website/source/docs/providers/aws/r/app_cookie_stickiness_policy.html.markdown]: @radeksimko the validation rules for this are not actually noted anywhere

http://docs.aws.amazon.com/cli/latest/reference/elb/create-app-cookie-stickiness-policy.html

just says:

```
--policy-name (string)

The name of the policy being created. This name must be unique within the set of policies for this load balancer.
```

I am not sure where this error comes from. Even the go-sdk doesn't seem to list the specific error you are getting:

https://github.com/aws/aws-sdk-go/blob/master/apis/elasticloadbalancing/2012-06-01/api-2.json#L178

Comment (radeksimko) [inline website/source/docs/providers/aws/r/app_cookie_stickiness_policy.html.markdown]: Hm, you're right, AWS docs are not always matching with the reality. I will open a support case, encourage them to update it. I've been already reporting some RDS validation, which was out of date the other day.

Until they document it, I think we should just validate against what the error says (i.e. alphanumeric + dash). Feel free to test certain edge cases, like two dashes, dashes at either end of the name, but I'd guess these will be allowed, based on the error message.

Comment (stack72) [inline website/source/docs/providers/aws/r/app_cookie_stickiness_policy.html.markdown]: kk, will update the PR with the Validation :)


-- 
Paul
Sent with Airmail

On 19 September 2015 at 19:55:51, Radek Simko (notifications@github.com) wrote:

In website/source/docs/providers/aws/r/app_cookie_stickiness_policy.html.markdown:

>  }
>   
>  resource "aws_app_cookie_stickiness_policy" "foo" {
> -    name = "foo_policy"
> -    load_balancer = "${aws_elb.lb.id}"
> -    lb_port = 80
> -    cookie_name = "MyAppCookie"
> +  name = "foo_policy"
Hm, you're right, AWS docs are not always matching with the reality. I will open a support case, encourage them to update it. I've been already reporting some RDS validation, which was out of date the other day.

Until they document it, I think we should just validate against what the error says (i.e. alphanumeric + dash). Feel free to test certain edge cases, like two dashes, dashes at either end of the name, but I'd guess these will be allowed, based on the error message.

Comment (radeksimko) [inline website/source/docs/providers/aws/r/app_cookie_stickiness_policy.html.markdown]: FYI: I just opened the support ticket, here's a copy: https://gist.github.com/radeksimko/f01168a61cbaa222b16c

**Comment by stack72** [comment]
@radeksimko this is now working as expected - i was able to reproduce this with the acceptance tests. I will open a subsequent PR to add a func that allows an acceptance test for throwing the correct validation errors :)

**Comment by radeksimko** [comment]
Thank you for your patience here, @stack72 

This LGTM now :+1:
